chain/consensus/signatures.go

```diff
@@ -18,34 +18,41 @@ import (
 // must be recognized by the registered verifier for the signature type.
 func AuthenticateMessage(msg *types.SignedMessage, signer address.Address) error {
 	var digest []byte
+	signatureType := msg.Signature.Type
+	signatureCopy := msg.Signature
 
-	typ := msg.Signature.Type
-	switch typ {
+	switch signatureType {
 	case crypto.SigTypeDelegated:
-		txArgs, err := ethtypes.EthTxArgsFromUnsignedEthMessage(&msg.Message)
+		signatureCopy.Data = make([]byte, len(msg.Signature.Data))
+		copy(signatureCopy.Data, msg.Signature.Data)
+		ethTx, err := ethtypes.EthTransactionFromSignedFilecoinMessage(msg)
```

Comment:
Coming back to this again I'm having some paranoid thoughts.  Does this PR open up forking as written?  
We release this code before nv23, someone comes in with a legacy Tx, sends it over to an SP with updated software.  That SP includes in a block.  Now an SP who is not updated yet reads the block.  It has this weird signature with one extra byte.  It doesn't validate per its code.  SP rejects block.

I could be off base and missing some subtlty that prevents this.  Just want to think it over with you before merge.

Reply:
I agree with @arajasek and @ZenGround0 that this PR would definitely create consensus issues if some nodes are able to execute a transaction and other nodes would reject it. Was there any discussion in the FIP process around how to handle the consensus breaking aspects of supporting a new transaction type?

Reply:
@snissn @ZenGround0 I have update this PR to only accept legacy txs if the node is on NV23. If the Node still hasn't upgraded to NV23, it will fail to accept legacy txns. 

I have also added an itest for it. It's in test `TestLegacyEIP155ValueTransferValidSignatureFailsNV22`. I think this should suffice. Let me know if you have any other concerns.